quic/s2n-quic-platform/src/io/testing.rs

```diff
@@ -296,14 +306,14 @@ impl Io {
             },
             address,
             on_socket,
-            max_mtu,
```

Comment:
Do we need to validate that the configuration is correct before using it everywhere?

Reply:
I'll handle that in the builder